### PR TITLE
93012 Replace back links with DS component

### PR DESCRIPTION
**Pull request by vbahinwillit**
**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

If the folder you changed contains a `manifest.json`, search for its `entryName` in the content-build [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) (the `entryName` there will match).

If an entry for this folder exists in content-build and you are:
1. **Deleting a folder**:
   1. First search `vets-website` for _all_ instances of the `entryName` in your `manifest.json` and remove them in a separate PR. Look particularly for references in `src/applications/static-pages/static-pages-entry.js` and `src/platform/forms/constants.js`. _**If you do not do this, other applications will break!**_
      - _Add the link to your merged vets-website PR here_
   2. Then, Delete the application entry in [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json) and merge that PR **before** this one
      - _Add the link to your merged content-build PR here_

2. **Renaming or moving a folder**: Update the entry in the [registry.json](https://github.com/department-of-veterans-affairs/content-build/blob/main/src/applications/registry.json), but do not merge it until your vets-website changes here are merged. The content-build PR must be merged immediately after your vets-website change is merged in to avoid CI errors with content-build (and Tugboat).

### :warning: TeamSites :warning:
Examples of a TeamSite: https://va.gov/health and https://benefits.va.gov/benefits/. This scenario is also referred to as the "injected" header and footer. You can reach out in the `#sitewide-public-websites` Slack channel for questions.

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary
This PR replaces back link with the design systems VaLink component.

## Related issue(s)
- _Link to ticket created in va.gov-team repo_
department-of-veterans-affairs/va.gov-team#93012

## Testing done
Unit and e2e

## Screenshots
N/A

## What areas of the site does it impact?
VAOS

## Acceptance criteria
All tests should pass

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/vaos/tests/e2e/page-objects/DateTimeSelectPageObject.js [hunk 1]
```diff
@@ -27,9 +27,7 @@ export class DateTimeSelectPageObject extends PageObject {
   assertUrl() {
     cy.url().should('include', '/date-time');
     cy.wait('@v2:get:slots');
-    cy.get('va-loading-indicator').as('spinner');
-    cy.get('@spinner').should('not.exist');
-    cy.axeCheckBestPractice();
```
Comment (JunTaoLuo) [inline src/applications/vaos/tests/e2e/page-objects/DateTimeSelectPageObject.js:32]: Hmmm what's the reason for these updates?

**Review by JunTaoLuo — APPROVED** [review]
Changes look good! I played around with the changes locally and it seems like everything is working as expected. What was the issue you mentioned with reloads you mentioned at stand-up? I didn't seem to hit it 🤔 

Also I have a bit of a UX question for @outerpress which is probably out of scope for this ticket but on the following page, I now see 3 different kinds of icons:
image
1. The new back link with the left arrow
2. The `<<` on the "Back" button
3. The `>>` on the "Continue" button which looks different than the `<<`?

I noticed that there were [plans](https://github.com/department-of-veterans-affairs/va.gov-team/issues/84550) to remove the "Back" button. Do you know if that's still the case? If not, do we need to update the buttons (back and continue) to use the new arrow icons?

File: src/applications/vaos/tests/e2e/page-objects/DateTimeSelectPageObject.js (hunk 1 above)
Comment (JunTaoLuo) [reply src/applications/vaos/tests/e2e/page-objects/DateTimeSelectPageObject.js:32]: Ah I see, these were for e2e test failures. It's bizarre but why would changing the back links cause these tests to fail? 

Also tests seem to pass locally for me even if `cy.axeCheckBestPractice();` but maybe that's related to the recent [scss change](https://github.com/department-of-veterans-affairs/vets-website/commit/eea487c91d65f8cb42f24d295375dae2ec142085)?

Comment (JunTaoLuo) [reply src/applications/vaos/tests/e2e/page-objects/DateTimeSelectPageObject.js:32]: Just pretend I'm talking to myself here, none of this blocks the PR from being merged btw.

**Review by CBonade — APPROVED** [review]

File: src/applications/vaos/tests/e2e/page-objects/DateTimeSelectPageObject.js (hunk 1 above)
Comment (vbahinwillit) [reply src/applications/vaos/tests/e2e/page-objects/DateTimeSelectPageObject.js:32]: No worries. Good questions.

1st Question: There is an option to use the [VaLink](https://design.va.gov/storybook/?path=/docs/components-va-link--docs#with-router-link-support) component, as opposed to the va-link web component when React routing is needed. This fixed the page loading issue I was seeing.

2nd Question: The removal of the check for the spinner and axe check fixed the flaky test issue I was seeing. I was seeing 2 separate issues: The va-loading-indicator was not found and a 508 error for color contrast. The root cause was a timing issue. The 1st issue only seemed to happen for the calendar-dead-ends Cypress tests only. The 2nd issue happens when the loading indicator is still running when the axe check runs.

Comment (JunTaoLuo) [reply src/applications/vaos/tests/e2e/page-objects/DateTimeSelectPageObject.js:32]: Thanks, it all makes a lot of sense!

**Review by jenniemc — APPROVED** [review]
LGTM!

**Pull request merged**